ext/bcmath/libbcmath/src/div.c

```diff
@@ -74,121 +75,109 @@ static void _one_mult(unsigned char *num, size_t size, int digit, unsigned char
+	unsigned int norm;
 
 	/* Test for divide by zero. */
 	if (bc_is_zero(n2)) {
 		return false;
 	}
 
-	/* Test for divide by 1.  If it is we must truncate. */
-	if (n2->n_scale == 0) {
-		if (n2->n_len == 1 && *n2->n_value == 1) {
-			qval = bc_new_num (n1->n_len, scale);
-			qval->n_sign = (n1->n_sign == n2->n_sign ? PLUS : MINUS);
-			memset (&qval->n_value[n1->n_len],0,scale);
-			memcpy (qval->n_value, n1->n_value, n1->n_len + MIN(n1->n_scale,scale));
-			bc_free_num (quot);
-			*quot = qval;
-		}
+	/* Test for divide by 1. If it is we must truncate. */
+	if (n2->n_scale == 0 && n2->n_len == 1 && *n2->n_value == 1) {
+		qval = bc_new_num (n1->n_len, scale);
+		qval->n_sign = (n1->n_sign == n2->n_sign ? PLUS : MINUS);
+		memset(&qval->n_value[n1->n_len], 0, scale);
+		memcpy(qval->n_value, n1->n_value, n1->n_len + MIN(n1->n_scale, scale));
+		bc_free_num (quot);
+		*quot = qval;
 	}
 
 	/* Set up the divide.  Move the decimal point on n1 by n2's scale.
 	   Remember, zeros on the end of num2 are wasted effort for dividing. */
 	scale2 = n2->n_scale;
-	n2ptr = (unsigned char *) n2->n_value+n2->n_len+scale2-1;
-	while ((scale2 > 0) && (*n2ptr-- == 0)) {
+	n2ptr = (unsigned char *) n2->n_value + n2->n_len + scale2 - 1;
+	while ((scale2 > 0) && (*n2ptr == 0)) {
 		scale2--;
+		n2ptr--;
 	}
 
 	len1 = n1->n_len + scale2;
 	scale1 = n1->n_scale - scale2;
-	if (scale1 < scale) {
-		extra = scale - scale1;
-	} else {
-		extra = 0;
-	}
-	num1 = (unsigned char *) safe_emalloc (1, n1->n_len+n1->n_scale, extra+2);
-	memset (num1, 0, n1->n_len+n1->n_scale+extra+2);
-	memcpy (num1+1, n1->n_value, n1->n_len+n1->n_scale);
+	extra = MAX(scale - scale1, 0);
+
+	num1 = (unsigned char *) safe_emalloc(1, n1->n_len + n1->n_scale, extra + 2);
+	memset(num1, 0, n1->n_len + n1->n_scale + extra + 2);
+	memcpy(num1 + 1, n1->n_value, n1->n_len + n1->n_scale);
 
 	len2 = n2->n_len + scale2;
-	num2 = (unsigned char *) safe_emalloc (1, len2, 1);
-	memcpy (num2, n2->n_value, len2);
-	*(num2+len2) = 0;
+	num2 = (unsigned char *) safe_emalloc(1, len2, 1);
+	memcpy(num2, n2->n_value, len2);
+	*(num2 + len2) = 0;
 	n2ptr = num2;
 	while (*n2ptr == 0) {
 		n2ptr++;
 		len2--;
 	}
 
 	/* Calculate the number of quotient digits. */
-	if (len2 > len1+scale) {
-		qdigits = scale+1;
-		zero = true;
+	if (len2 > len1 + scale || len2 > len1) {
+		qdigits = scale + 1;
 	} else {
-		zero = false;
-		if (len2 > len1) {
-			/* One for the zero integer part. */
-			qdigits = scale+1;
-		} else {
-			qdigits = len1-len2+scale+1;
-		}
+		qdigits = len1 - len2 + scale + 1;
 	}
 
 	/* Allocate and zero the storage for the quotient. */
-	qval = bc_new_num (qdigits-scale,scale);
-	memset (qval->n_value, 0, qdigits);
+	qval = bc_new_num (qdigits - scale, scale);
+	memset(qval->n_value, 0, qdigits);
 
 	/* Allocate storage for the temporary storage mval. */
 	mval = (unsigned char *) safe_emalloc(1, len2, 1);
 
 	/* Now for the full divide algorithm. */
-	if (!zero) {
+	if (len2 <= len1 + scale) {
```

Comment:
I'm not sure this code is more legible now

Reply:
Reverted this part of the code.